amida-tech/orange-api
# Change clinician boolean in user to role enum

#### What does this PR do?
Previously the user had a boolean field called "clinician" that was a bool.
Now it is an enum called "role" with two possible values "clinician" and "user".

Also, the user model now has a new field "npi" that can be set via a PUT only if the user has the role "clinician".

Finally, we now have a npi lookup endpoint `npi/:npi`

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/ORANGE-203
https://jira.amida-tech.com/browse/ORANGE-204
https://jira.amida-tech.com/browse/ORANGE-123

#### How should this be manually tested?
---
Using Postman (or similar) register a new user as described [here](http://amida-tech.github.io/orange-api/#users-user-post):
* try without setting the "role" field
  * should return `invalid_role`
* set the role field to "any random string"
  * should return `invalid_role`
* set the role to "user"
  * should return new user with role: `user`
* set the role to "clinician"
  * should return new user with role: `clinician`
---
Try to modify an existing user as described [here](http://amida-tech.github.io/orange-api/#users-user-put)
* on a user with role: "user"
  * try to update the `npi`
  * it shouldn't do anything, return user still should not have an npi
* on a user with role: "clinician"
  * try to set the npi to a ten digit number
    * should succeed with npi in response
  * try to set npi to empty string or null
    * should succeed with blank npi in response
  * try to set npi to some random string
    * should fail
---
* use postman to GET `/npi/1245319599`
  * should succeed with valid info
* GET `/npi/13`
  * should return empty `{}`

#### Background/Context
Using an enum for the role of a user allows us greater flexibility s.t. clinician and non-clinician are not the only roles. For now the only roles in the enum are "clinician" and "user" but we can add more easily.

This doesn't include setting the npi of a clinician user when creating the user.
